### PR TITLE
Create tests for _fft_r2c | test(torchlib)

### PR DESCRIPTION
**Pull request by fatcat-z**
Update test for _fft_r2c op and fixed its implementation.

### CODE REVIEW

File: onnxscript/function_libs/torch_lib/ops/fft.py [hunk 1]
```diff
@@ -95,11 +95,16 @@ def _fftn_onnx(
...
 
-    for dim_ in dims:
-        if dim_ >= 0:
-            # Add 1 to account for the batch dimension when counting axes from the left
-            dim_ = dim_ + 1
-        transformed = op.DFT(transformed, axis=dim_, inverse=inverse, onesided=onesided)
+    new_dims = [dim_ + 1 if dim_ >= 0 else dim_ for dim_ in dims]
+
+    loop_count = len(new_dims) - 1
+    for i in range(loop_count):
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/fft.py:101]:
```suggestion
    for dim in new_dims[:-1]:
```

File: onnxscript/function_libs/torch_lib/ops/fft.py [hunk 2]
```diff
@@ -95,11 +95,16 @@ def _fftn_onnx(
...
+        transformed = op.DFT(transformed, axis=new_dims[i], inverse=inverse, onesided=False)
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/fft.py:102]:
```suggestion
        transformed = op.DFT(transformed, axis=dim, inverse=inverse, onesided=False)
```

File: onnxscript/function_libs/torch_lib/ops/fft.py [hunk 3]
```diff
@@ -95,11 +95,16 @@ def _fftn_onnx(
...
+
+    if onesided:
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/fft.py:104]: I would add comment to explain this logic. Thanks!

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py [hunk 4]
```diff
@@ -465,6 +465,12 @@ def _where_input_wrangler(
         trace_only=True,
         complex=True,
     ),
+    TorchLibOpInfo(
+        "ops.aten._fft_r2c",  # Custom from extra_opinfo
+        fft_ops.aten__fft_r2c,
+        tolerance={torch.float64: (2e-6, 2e-6), torch.float32: (3e-3, 3e-5)},
```
Comment (fatcat-z) [inline onnxscript/tests/function_libs/torch_lib/ops_test_data.py:471]:
```suggestion
        tolerance={torch.float64: (2e-6, 2e-6), torch.float32: (3e-3, 2e-5)},
```

File: onnxscript/tests/function_libs/torch_lib/extra_opinfo.py [hunk 5]
```diff
@@ -214,25 +213,31 @@ def sample_inputs__fft_c2c(self, device, dtype, requires_grad=False, **_):
...
+            shapes[0],
             device=device,
             low=low,
             high=high,
             dtype=dtype,
             requires_grad=requires_grad,
         )
 
+    return oned_tensor, nd_tensor
+
```
Comment (fatcat-z) [inline onnxscript/tests/function_libs/torch_lib/extra_opinfo.py:236]:
```suggestion


```

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py [hunk 6]
```diff
@@ -465,6 +465,12 @@ def _where_input_wrangler(
         trace_only=True,
         complex=True,
     ),
+    TorchLibOpInfo(
+        "ops.aten._fft_r2c",  # Custom from extra_opinfo
+        fft_ops.aten__fft_r2c,
+        tolerance={torch.float64: (2e-6, 2e-6), torch.float32: (3e-3, 3e-4)},
```
Comment (fatcat-z) [inline onnxscript/tests/function_libs/torch_lib/ops_test_data.py:471]:
```suggestion
        tolerance={torch.float64: (2e-6, 2e-6), torch.float32: (3e-2, 3e-4)},
```

**Comment by justinchuby** [comment]
Feel free to merge this first and start a new PR if you like

**Review by justinchuby — APPROVED** [review]

**Pull request merged**